src/main/java/org/truffleruby/core/symbol/SymbolTable.java

```diff
@@ -147,4 +149,11 @@ public static String checkClassVariableName(
         return name;
     }
 
+    @TruffleBoundary(transferToInterpreterOnException = false)
+    public RubySymbol[] allSymbols() {
+        final Collection<RubySymbol> allSymbols = symbolMap.values();
+        // allSymbols is a concrete collection not a view
+        return allSymbols.toArray(new RubySymbol[allSymbols.size()]);
```

Comment:
I consider `collection.toArray(new SomeType[collection.size()]);` an anti-pattern nowadays because:
* It's more verbose than `collection.toArray(new SomeType[0]);`.
* It can end up with `null` in it if the collection can shrink concurrently like here.
* It doesn't seem more efficient than `[0]`.

I'll try to add a CheckStyle rule to check that.

That said, for this case the simplest to use `allSymbols.toArray()` since anyway we need a `Object[]`, not a `RubySymbol[]`.

Reply:
For reference the previous code did it like:
https://github.com/oracle/truffleruby/blob/release/graal-vm/20.1/src/main/java/org/truffleruby/core/symbol/SymbolTable.java#L159-L162
https://github.com/oracle/truffleruby/blob/release/graal-vm/20.1/src/main/java/org/truffleruby/core/symbol/SymbolNodes.java#L53-L56

Reply:
Doesn't `symbolMap.values()` return a private copy? I don't think it can shrink concurrently.

Reply:
You're right, `WeakValueCache#values()` is eager (due to the implementation detail that it needs to filter nulled references) and so doesn't present that race, in opposite to e.g., `ConcurrentHashMap#values()` which has the issue.

Reply:
Fixed.